language/move-prover/tests/sources/stdlib/modules/libra.move

```diff
@@ -15,12 +10,41 @@ module Libra {
         value: u64,
     }
 
+    spec module {
+         // This ghost variable is defined to have the true sum values of all instances of Token
+         global sum_of_token_values<Token>: num;
+    }
+
+    // maintain sum_of_tokens
+    spec struct T {
+        invariant pack sum_of_token_values<Token> = sum_of_token_values<Token> + value;
+        invariant unpack sum_of_token_values<Token> = sum_of_token_values<Token> - value;
+        invariant update sum_of_token_values<Token> = sum_of_token_values<Token> - old(value) + value;
```

Comment:
Notice that if pack/unpack is symmetric, you can also leave the update out -- it will be derived.

Reply:
Removed.